### PR TITLE
Merge/download attributes

### PR DESCRIPTION
**Pull request by ifoche**
This PR contains the following functionality:
- Download attributes
- Download ProgramAtrributesValues
- Download DataElementAtrributesValues
- Download ProgramstageAtrributesValues

### CODE REVIEW

**Comment by vgarciabnz** [comment]
Conflicts resolved in https://github.com/dhis2/dhis2-android-sdk/pull/1475

File: core/src/main/java/org/hisp/dhis/android/core/attribute/Attribute.java [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+    @JsonProperty()
+    public abstract Boolean organisationUnitGroupSetAttribute();
+
+    @Nullable
+    @JsonProperty()
+    public abstract Boolean optionAttribute();
+    
+
+    public static Builder builder() {
+        return new $$AutoValue_Attribute.Builder();
```
Comment (aimiebell) [inline core/src/main/java/org/hisp/dhis/android/core/attribute/Attribute.java:187]: I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.

File: core/src/main/java/org/hisp/dhis/android/core/attribute/DataElementAttributeValueLink.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+    @Nullable
+    public abstract String value();
+
+    public static DataElementAttributeValueLink create(Cursor cursor) {
+        return AutoValue_DataElementAttributeValueLink.createFromCursor(cursor);
+    }
+
+    public static Builder builder() {
+        return new $$AutoValue_DataElementAttributeValueLink.Builder();
```
Comment (aimiebell) [inline core/src/main/java/org/hisp/dhis/android/core/attribute/DataElementAttributeValueLink.java:57]: I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.

File: core/src/main/java/org/hisp/dhis/android/core/attribute/ProgramAttributeValueLink.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+    @Nullable
+    public abstract String value();
+
+    public static ProgramAttributeValueLink create(Cursor cursor) {
+        return AutoValue_ProgramAttributeValueLink.createFromCursor(cursor);
+    }
+
+    public static Builder builder() {
+        return new $$AutoValue_ProgramAttributeValueLink.Builder();
```
Comment (aimiebell) [inline core/src/main/java/org/hisp/dhis/android/core/attribute/ProgramAttributeValueLink.java:57]: I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.

File: core/src/main/java/org/hisp/dhis/android/core/attribute/ProgramStageAttributeValueLink.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+    @Nullable
+    public abstract String value();
+
+    public static ProgramStageAttributeValueLink create(Cursor cursor) {
+        return AutoValue_ProgramStageAttributeValueLink.createFromCursor(cursor);
+    }
+
+    public static Builder builder() {
+        return new $$AutoValue_ProgramStageAttributeValueLink.Builder();
```
Comment (aimiebell) [inline core/src/main/java/org/hisp/dhis/android/core/attribute/ProgramStageAttributeValueLink.java:57]: I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.

File: core/src/main/java/org/hisp/dhis/android/core/dataelement/DataElement.java [hunk 5]
```diff
@@ -92,6 +97,11 @@ public String categoryComboUid() {
     @Nullable
     public abstract String fieldMask();
 
+    @Nullable
+    @JsonProperty()
+    @ColumnAdapter(IgnoreAttributeValuesListAdapter.class)
+    public abstract List<AttributeValue> attributeValues();
+
     public static Builder builder() {
         return new $$AutoValue_DataElement.Builder();
```
Comment (aimiebell) [inline core/src/main/java/org/hisp/dhis/android/core/dataelement/DataElement.java:106]: I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.

**Pull request closed without merging**